### PR TITLE
Retain dialogs on screen rotation - 'What's New' help option, and chan…

### PR DESCRIPTION
**Pull request by nmurali94**
#### What is it?
- [x] Bug fix (user facing)
- [ ] Feature (user facing)
- [ ] Code base improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Fixes the following issue(s)

- Prevent popups from being dismissed on screen rotation like  'What's New' help option and others.
These are the first steps towards consolidating how some of these popups are created. The current implementation works for items in regular streams, channels and subscriptions.

#### Testing apk

[app-debug.zip](https://github.com/TeamNewPipe/NewPipe/files/5171910/app-debug.zip)

#### Agreement
- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.

### CODE REVIEW

File: app/src/main/java/org/schabi/newpipe/info_list/InfoItemDialog.java [hunk 1]
```diff
@@ -1,54 +1,86 @@
...
+public class InfoItemDialog  {
+
+    private final ItemDialogFragment idf;
+
+    public InfoItemDialog(@NonNull final String title,
+                          final String additionalInfo,
                           @NonNull final String[] commands,
-                          @NonNull final DialogInterface.OnClickListener actions) {
-        this(activity, commands, actions, info.getName(), info.getUploaderName());
+                          @NonNull final OnClickListener actions) {
```
Comment (nmurali94) [inline app/src/main/java/org/schabi/newpipe/info_list/InfoItemDialog.java:26]: These are the first steps towards consolidating how some of these popups are created. 
The current implementation works for items in regular streams, channels and subscriptions.

**Comment by avently** [comment]
It's hard to imaging that this pr will ever be merged
By making retainedInstance you just ruined every fragment's lifecycle. So the fragments will not be recreated on configuration change. 
And maybe you can disagree, but there is no any problem with a situation when you have missing dialogs on orientation change. Nothing bad happens.

**Comment by nmurali94** [comment]
I think it's always good if we can keep screen elements after rotation. By putting the popups in the dialog fragment we've got that even if we potentially lose some actions in the popup (the parent fragment is destroyed)

I can certainly understand that retainInstance isn't desirable, it was lazy solution. I'll spend some more time when I get the chance to see if I work around it.

**Comment by Stypox** [comment]
@nmurali94 could you explain what you did technically in this PR? I am worried by what @avently said, but do not understand what it means ;-)

**Comment by mhmdanas** [comment]
@nmurali94 I'm closing this PR, but please make a comment when you want to resume work on this so I would re-open it. Thanks!

**Pull request closed without merging**